core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -351,22 +352,28 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
     val e2 = intercept[ExecutionException] { adminClient.describeTopics(Set(topic).asJava).all().get() }
     assertTrue("Unexpected exception " + e2.getCause, e2.getCause.isInstanceOf[TopicAuthorizationException])
 
-    // Verify that consumer manually assigning both authorized and unauthorized topic doesn't consume from either
+    // Verify that consumer manually assigning both authorized and unauthorized topic doesn't consume
+    // from the unauthorized topic and throw; since we can now return data during the time we are updating
+    // metadata / fetching positions, it is possible that the authorized topic record is returned during this time.
     consumer.assign(List(tp, tp2).asJava)
     sendRecords(producer, numRecords, tp2)
+    var topic2RecordConsumed = false
```

Comment:
As explained in the above comment, with this change we may still return data for `topic2` if only `topic` is not authorized, so this test needs to be updated.

Reply:
I wonder if we really need to be testing with two separate topics here. We already have a hard time with the flakiness of this test.

Reply:
I think it is sound to have two topics here since topic1 is for read/write to a topic without ACL restrictions, and topic2 is for read/write to a topic with ACL restriction, and then granting the permission.

I hope I've figured out all the flakiness of this one after spending 10+ hours on it already :P and actually one of the flakiness did lead to some real issue here, so I'd prefer we keep it as is.